src/core/qgstiledownloadmanager.h

```diff
@@ -0,0 +1,292 @@
+  public:
+    ~QgsTileDownloadManagerReply();
+
+    //! Returns whether the reply has already finished (with success/failure)
+    bool hasFinished() const { return mHasFinished; }
+    //! Returns binary data returned in the reply (only valid when already finished)
+    QByteArray data() const { return mData; }
+    //! Returns error code (only valid when already finished)
+    QNetworkReply::NetworkError error() const { return mError; }
+    //! Returns error string (only valid when already finished)
+    QString errorString() const { return mErrorString; }
+
+    //! Returns the original request for this reply object
+    QNetworkRequest request() const { return mRequest; }
+
+  signals:
+    //! Emitted when the reply has finished (either with a success or with a failure)
+    void finished();
+
+  private slots:
+    void requestFinished( QByteArray data, QNetworkReply::NetworkError error, const QString &errorString );
+
+  private:
+    QgsTileDownloadManagerReply( QgsTileDownloadManager *manager, const QNetworkRequest &request );
+
+    friend class QgsTileDownloadManager;  // allows creation of new instances from the manager
+
+  private:
+    //! "parent" download manager of this reply
+    QgsTileDownloadManager *mManager = nullptr;
+    QNetworkRequest mRequest;
+    bool mHasFinished = false;
+    QByteArray mData;
+    QNetworkReply::NetworkError mError = QNetworkReply::NoError;
+    QString mErrorString;
+};
+
+
+/// @cond PRIVATE
+
+/**
+ * \ingroup core
+ * Reply object that is used in the worker thread of the tile download manager.
+ * \note This class is not a part of public API
+ */
+class QgsTileDownloadManagerReplyWorkerObject : public QObject
+{
+    Q_OBJECT
+  public:
+    QgsTileDownloadManagerReplyWorkerObject( QgsTileDownloadManager *manager, const QNetworkRequest &request )
+      : mManager( manager ), mRequest( request ) {}
+
+  public slots:
+    void replyFinished();
+
+  signals:
+    void finished( QByteArray data, QNetworkReply::NetworkError error, const QString &errorString );
+
+  private:
+    //! "parent" download manager of this worker object
+    QgsTileDownloadManager *mManager = nullptr;
+    QNetworkRequest mRequest;
+};
+
+
+/**
+ * \ingroup core
+ * Worker object that is used in the worker thread of the tile download manager.
+ * \note This class is not a part of public API
+ */
+class QgsTileDownloadManagerWorker : public QObject
+{
+    Q_OBJECT
+
+  public:
+    //! Creates the worker
+    QgsTileDownloadManagerWorker( QgsTileDownloadManager *manager, QObject *parent = nullptr );
+
+    void startIdleTimer();
+
+  public slots:
+    void queueUpdated();
+    void idleTimerTimeout();
+
+  signals:
+    void requestFinished( QString url, QByteArray data );
+
+  private:
+    void quitThread();
+
+  private:
+    //! "parent" download manager of this worker
+    QgsTileDownloadManager *mManager = nullptr;
+    //! Timer used to delete the worker thread if thread is not doing anything for some time
```

Comment:
Timer used -> Timer is used

Reply:
I'm re-reading this but it seems written correctly to me...?

Reply:
small difference but i do not mind